pkg/aws/client.go

```diff
@@ -0,0 +1,48 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+)
+
+// Client provides the single API client to make operations call to aws services
+type Client struct {
+	ec2 *ec2.Client
+}
+
+// Clients are a map between aws profile and its aws client
+type Clients map[string]*Client
+
+type ServiceEndpoint struct {
+	ServiceID     string
+	URL           string
+	SigningRegion string
+}
+
+type awsConfigOpts []func(*config.LoadOptions) error
+
+func LoadConfig(ctx context.Context, credsFilePath string, opts ...awsConfigOpts) (aws.Config, error) {
```

Comment:
This signature is a little odd, its forcing consumers to build option sets rather than simply passing the individual options. I think that complexity should lie in the option itself, not this signature. 

The only options we have at present is `WithSnow()` which is an option set where everything in the set is required. We can model that while still keeping the `LoadConfig` simple.

```go
type AwsConfigOpt func(*config.LoadOptions) error

func LoadConfig(..., opts AwsConfigOpt) (aws.Config, error) { ... }

func AwsConfigOptSet(opts ...AwsConfigOpt) AwsConfigOpt {
  return func(conf *config.LoadOptions) error {
    for _, opt := range opts {
      if err := opt(conf); err != nil {
        return err
      }
    }

    return nil
  }
}

func WithSnow(deviceIP string, certsFile string) AwsConfigOpt {
	return AwsConfigOptSet(
		withCustomCABundleFile(certsFile),
		config.WithSharedConfigProfile(deviceIP),
		config.WithEndpointResolverWithOptions(SnowEndpointResolver(deviceIP)),
	)
}


```

Reply:
A secondary note, if we expect consumers of the  package to interact with a type its typically best to export that type. It really helps with testability beacuse you can test options modify the `config.LoadOptions` as you expect. This is why, in the example above, I changed the config opt type to be exported.